contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py

```diff
@@ -71,6 +71,11 @@ def export_metrics(self, metrics):
         for batch in batched_envelopes:
             batch = self.apply_telemetry_processors(batch)
             result = self._transmit(batch)
+            # If statsbeat exporter and received signal to shutdown
+            if self._is_stats_exporter() and result == -2:
```

Comment:
_statsbeat_failed_to_ingest above can the counter and here just check if the counter is >= 3. -2 seems so random.

Reply:
`_statsbeat_failed_to_ingest ` is a private function used only within `transport` to handle the count as well as determining whether it is reached. The returning of the result code back to the exporter is by design. I agree the codes are a bit random (-1, -2) but changing them can be part of a different PR.